src/types/levelState.ts

```diff
@@ -0,0 +1,107 @@
+// All levels together in a single state object
```

Comment:
I think it's going to take quite a bit of testing to get to something that's actually fun, so I think it's worthwhile to make things as configurable as possible.